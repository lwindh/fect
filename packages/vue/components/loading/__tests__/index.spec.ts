import { mount } from '@vue/test-utils'
import Loading from '..'

const Wrapper = {
  components: {
    [Loading.name]: Loading,
  },
  data() {
    return {
      type: 'default',
      size: 'medium',
      color: '',
      loadType: 'default',
    }
  },
  template: `
   <div class="container">
     <fe-loading :type="type" 
     :size="size" :color="color" 
     :load-type="loadType" :type-data="type" :type-size="size"/>
   </div>
  `,
}

describe('Loading', () => {
  it('should be rendera as a element', () => {
    const wrapper = mount(Loading)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('component props should be work correctly', async () => {
    const wrapper = mount(Wrapper)
    const loadEl = wrapper.find('.fect-loading')
    expect(loadEl.attributes('type-data')).toBe('default')
    expect(loadEl.attributes('type-size')).toBe('medium')
    await wrapper.setData({ type: 'success', size: 'small' })
    expect(loadEl.attributes('type-data')).toBe('success')
    expect(loadEl.attributes('type-size')).toBe('small')
    await wrapper.setData({ color: 'pink' })
    const el = wrapper.find('i')
    expect(el.attributes('style')).toBe('background: pink;')
    await wrapper.setData({ loadType: 'cube' })
    expect(wrapper.findAll('.loading__cube')).toBeTruthy()
    await wrapper.setData({ loadType: 'wave' })
    expect(wrapper.findAll('.loading__wave')).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
