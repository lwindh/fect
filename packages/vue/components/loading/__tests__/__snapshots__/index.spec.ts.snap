// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component props should be work correctly 1`] = `
<div class="container">
  <div class="fect-loading" type-data="success" type-size="small"><span class="loading"><i class="loading__wave success small" style="background: pink;"></i><i class="loading__wave success small" style="background: pink;"></i><i class="loading__wave success small" style="background: pink;"></i><i class="loading__wave success small" style="background: pink;"></i><i class="loading__wave success small" style="background: pink;"></i></span></div>
</div>
`;

exports[`Loading should be rendera as a element 1`] = `<div class="fect-loading"><span class="loading"><i class="loading__default default medium"></i><i class="loading__default default medium"></i><i class="loading__default default medium"></i></span></div>`;
